### DISCUSSION TITLE
Data fetching using sever action

### DISCUSSION THREAD

**Discussion by abhishekmardiya** [vercel/next.js#62272] [App Router] [answered]
Is it considered good practice to fetch data using server actions?

**Comment by Sodiumchloridy** [comment, marked as answer]
From the NextJS team themselves,

>"Whenever possible, we recommend fetching data on the server with Server Components."
> https://nextjs.org/docs/app/building-your-application/data-fetching/patterns

and server actions are one of the ways to fetch data (along with fetch etc) on the server, so I would say yes. Basically, it reduces the loading time needed to fetch data from multiple places on the client side, since it will need to fetch from the server anyways, and the server can cache data too which is even better.
